tests/pods_test.go

```diff
@@ -37,6 +38,13 @@ type Event struct {
 func TestPod(testT *testing.T) {
 	// https://stack-rox.atlassian.net/browse/ROX-6631
 	// - the process events expected in this test are not reliably detected.
+
+	if os.Getenv("COLLECTION_METHOD") == "NO_COLLECTION" {
+		log.Infof("Skipping TestPod because env var \"COLLECTION_METHOD\" is set to \"NO_COLLECTION\"\n" +
+			"This is expected to only happen when Sensor version is 3.74.x (support exception within compatibility tests)")
+		return
+	}
```

Comment:
I see two issues here.

1. If we skip a test, we should use `t.Skip` method so that this is clearly marked as skipped.
2. This test checks many things and only a part of those rely on collector being active. Could we maybe wrap the part where we assert on container events in an `if` block, so that it is only checked if the collection method is other than `NO_COLLECTION`?

Reply:
I could change it to only gate off parts of the test that don't check events, sure. That would make the test a lot smaller in that case, but perhaps it's better than nothing

Reply:
I am fine with disabling the tests as a solution, but please, let's use the `t.Skip` for that (see my comment above).

What we should check is whether:
1. The expected log line is missing due to the collector version extraction failing but the cluster works fine OR 
2. The expected log line is missing because the failing extraction of collector version prevents sensor from connecting to Central.

It is fine to conclude this PR when the skips are addressed and comments are added, but please let's go deeper in the comments, so that we understand why the collector version check fails and how it prevents the log line from appearing.

Reply:
For PodsTest specifically I have refactored it to not skip the whole test but only everything surrounding the event that does not appear when `NO_COLLECTION` is set, that way the rest of it is still tested, so I don't think `t.Skip` works here (unless we want to skip the whole test after all).

I'll add more comments to the TestTLSChallenge now 👍